modules/tinymce/src/core/main/ts/selection/NormalizeRange.ts

```diff
@@ -59,17 +64,22 @@ const hasContentEditableFalseParent = (body: HTMLElement, node: Node) => {
 // Walks the dom left/right to find a suitable text node to move the endpoint into
 // It will only walk within the current parent block or body and will stop if it hits a block or a BR/IMG
 const findTextNodeRelative = (dom: DOMUtils, isAfterNode: boolean, collapsed: boolean, left: boolean, startNode: Node): Optional<CaretPosition> => {
-  let lastInlineElement;
   const body = dom.getRoot();
-  let node;
   const nonEmptyElementsMap = dom.schema.getNonEmptyElements();
+  const parentNode = startNode.parentNode;
+  let lastInlineElement: Node | undefined;
+  let node: Node | undefined;
+
+  if (!parentNode) {
```

Comment:
Another check to handle the node potentially being detached/where it's parent node is null.